scripts/wamr_hourly.py

```diff
@@ -87,46 +81,7 @@ def main():
 
     args = parser.parse_args()
 
-    # Open up any resources that we need for the program
-
-    # Logging
-    log = setup_logging(args.log_level, args.log, args.error_email)
-    log.info('Starting WAMR rtd')
-
-    # Database connection
-    try:
-        engine = create_engine(args.connection_string)
-        Session = sessionmaker(engine)
-        sesh = Session()
-    except Exception as e:
-        log.critical('Error with Database connection', exc_info=True)
-        sys.exit(1)
-
-    # Output files
-    if args.error_file:
-        error_file = open(args.error_file, 'a')
-    else:
-        error_filename = 'wamr_errors_{}.csv'.format(datetime.strftime(
-                datetime.now(), '%Y-%m-%dT%H-%M-%S'))
-        error_file = open(os.path.join(args.cache_dir, error_filename), 'a')
-
-    if args.input_file:
-        with open(args.input_file) as f:
-            rows, fieldnames = file2rows(f, log)
-    else:  # FTP
-        rows, fieldnames = ftp2rows(args.ftp_server, args.ftp_dir, log)
-
-        if not args.cache_file:
-            args.cache_file = 'wamr_download_{}.csv'.format(datetime.strftime(
-                datetime.now(), '%Y-%m-%dT%H-%M-%S'))
-        with open(args.cache_file, 'w') as cache_file:
-            cache_rows(cache_file, rows, fieldnames)
-
-    log.info('{0} observations read into memory'.format(len(rows)))
-
-    # Hand the row off to the database processings/insertion part of the script
-    rows2db(sesh, rows, error_file, log, args.diag)
-
+    run(args)
```

Comment:
You can move this and the ``parse_args()`` call into the ``if __name__ == '__main__'`` conditional and get rid of the ``main()`` function altogether.